src/PhpWord/Writer/Word2007/Element/Table.php

```diff
@@ -103,8 +103,14 @@ private function writeRow(XMLWriter $xmlWriter, RowElement $row): void
         }
 
         // Write cells
-        foreach ($row->getCells() as $cell) {
-            $this->writeCell($xmlWriter, $cell);
+        $cells = $row->getCells();
+        if (count($cells) === 0) {
+            // issue 2505 - Word treats doc as corrupt if row without cell
+            $this->writeCell($xmlWriter, new CellElement());
+        } else {
+            foreach ($cells as $cell) {
+                $this->writeCell($xmlWriter, $cell);
+            }
         }
```

Comment:
```suggestion
        if (count($row->getCells()) === 0) {
            // Add a cell if the row has no cells (avoid corrupt file)
            $row->addCell();
        }
        foreach ($row->getCells() as $cell) {
            $this->writeCell($xmlWriter, $cell);
        }
```

Reply:
I'm not sure about this suggestion. The way I have coded it, there is no change to the user's document object after the write. With your suggestion, there will be a new cell added to the user's document object. This seems unexpected. If the user were to write, then add a cell to the row (perhaps user is preparing multiple documents), it would now be in the row's second cell rather than the first. Given that, do you still think this should be recoded?

Reply:
This would mutate the document object when writing out a file. As a user I would expect that the document object is not modified.